src/org/intellij/erlang/compilation/ErlangPrepareDependenciesCompileTask.java

```diff
@@ -128,161 +128,118 @@ private static ErlangModuleBuildOrderDescriptor getModuleBuildOrderInner(Module
     ErlangFacet erlangFacet = ErlangFacet.getFacet(module);
     List<String> globalParseTransforms = erlangFacet != null ? erlangFacet.getConfiguration().getParseTransforms() : ContainerUtil.<String>emptyList();
     buildOrder.myModuleName = module.getName();
-    buildOrder.myOrderedErlangModulePaths = getTopologicallySortedErlangModulePaths(ErlangModulesUtil.getErlangModules(module, false), globalParseTransforms);
-    buildOrder.myOrderedErlangTestModulePaths = getTopologicallySortedErlangModulePaths(ErlangModulesUtil.getErlangModules(module, true), ContainerUtil.<String>emptyList());
+    buildOrder.myOrderedErlangModulePaths = getTopologicallySortedErlangModulePaths(ErlangModulesUtil.getErlangModules(module, false),
+                                                                                    ErlangModulesUtil.getErlangHeaders(module, false),
+                                                                                    globalParseTransforms);
+    buildOrder.myOrderedErlangTestModulePaths = getTopologicallySortedErlangModulePaths(ErlangModulesUtil.getErlangModules(module, true),
+                                                                                        ErlangModulesUtil.getErlangHeaders(module, true),
+                                                                                        ContainerUtil.<String>emptyList());
```

Comment:
Again, these are not module paths anymore. Please, be consistent in naming. It severely improves code readability if things are named properly.